lib/base-compiler.js

```diff
@@ -103,6 +103,22 @@ class BaseCompiler {
         return exec.execute(compiler, args, options);
     }
 
+    getCompilerCacheKey(compiler, args, options) {
+        return {mtime: this.mtime, compiler, args, options};
+    }
+
+    async execCompilerCached(compiler, args, options) {
+        const key = this.getCompilerCacheKey(compiler, args, options);
+        let result = await this.env.compilerCacheGet(key);
+        if (!result) {
+            result = await exec.execute(compiler, args, options);
+            if (result.okToCache)
+                this.env.compilerCachePut(key, result);
```

Comment:
the cache put returns a Promise; I _think_ it's OK to ignore it to keep the put being async, but I'm not 100% sure.

Reply:
That behavior was copied from the caching bits in `compile`. Not sure if it was intentional there either, but it does appear to work.